src/components/HighContrastSelector/HighContrastSelector.tsx

```diff
@@ -0,0 +1,22 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import * as React from 'react';
+
+import 'HighContrastSelector.css';
+
+export interface HighContrastSelectorProps {
+  readonly regular: React.ReactNode;
+  readonly highContrast: React.ReactNode;
+}
+
+export default class HighContrastSelector extends React.Component<HighContrastSelectorProps> {
+  public render() {
+    const { regular, highContrast } = this.props;
+
+    return (
+      <>
+        <div className="y-high-contrast-selector--regular">{regular}</div>
+        <div className="y-high-contrast-selector--highContrast">{highContrast}</div>
+      </>
```

Comment:
Can you think of a way of not rendering both on the page, in case they're costly to render?

I was thinking something like:

```
<>
  <div className="y-high-contrast-selector__canary" ref={this.setRef} />
  {!this.state.isHighContrast && <div className="y-high-contrast-selector--regular">{regular}</div>}
  {this.state.isHighContrast && <div className="y-high-contrast-selector--highContrast">{highContrast}</div>}
</>
```

Then when `setRef` gets called, we check its `display` or some other attribute we set with CSS, and call `this.setState({ isHighContrast: true/false })`

Reply:
KISS then.